src/devices/cpu/t11/t11ops.hxx

```diff
@@ -268,6 +268,8 @@
 #define XOR_R(d)    int sreg, dreg, source, dest, result;     GET_SREG; source = REGW(sreg); GET_DW_##d; CLR_NZV; result = dest ^ source; SETW_NZ; PUT_DW_DREG(result)
 #define XOR_M(d)    int sreg, dreg, source, dest, result, ea; GET_SREG; source = REGW(sreg); GET_DW_##d; CLR_NZV; result = dest ^ source; SETW_NZ; PUT_DW_EA(result)
 
+/* test if insn is supported by the CPU */
+#define CHECK_IS(d) if (!(c_insn_set & (d))) { illegal(op); return; }
```

Comment:
This can produce weird effects if someone does something like:
```c++
	if (foo)
		CHECK_IS(IS_LEIS)
	else
		bar();
```

To make the macro work more like a single statement, wrap it in a dummy do/while loop:
```c++
#define CHECK_IS(d) do { if (!(c_insn_set & (d))) { illegal(op); return; } while (false)
```